dev/src/dev/toucan2_monitor.clj

```diff
@@ -42,6 +54,12 @@
 (defn start!
   "Start tracking queries."
   []
+  (tu.log/set-ns-log-level! *ns* :debug)
+  (reset! log-future (future
+                      (while true
+                        (let [{:keys [total-queries total-execution-time]} (summary)]
+                          (log/infof "Total queries: %d, Total execution time: %dms" total-queries total-execution-time)
+                          (Thread/sleep 1000)))))
```

Comment:
It's not ideal to sleep like this in a future, as futures come from a global thread pool that isn't that large, and this will tie up that thread.

Another option is to work directly with a Java `Thread`, or to use `core.async`.

Since this is just for dev, I think it's fine, but also thought its worth pointing out for future edge cases. Maybe put a comment there noting the issue and saying why it's OK 😄

Reply:
Just as a data point, I disagree with `core.async` mention. 🤣

Reply:
Haha I don't know what the temperature on that library is, I just saw we have the dependency. While channels are awesome in Go, I do remember there being a ton of rough edges in the Clojure version when I first used it? Didn't enjoy how opaque they were in the REPL, and hard to clean up.

Reply:
> While channels are awesome in Go

Can't disagree more, channels suck everywhere. :) Especially when the codebase is not your own it can be extremely difficult to understand where do values come from.

Reply:
use thread primitives here [28636f8](https://github.com/metabase/metabase/pull/45176/commits/28636f8d38f8731d5541416a74029109b1524808)